OpenRA.Mods.Common/Orders/OrderGenerator.cs

```diff
@@ -0,0 +1,35 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.Orders
+{
+	public abstract class OrderGenerator : IOrderGenerator
+	{
+		public IEnumerable<Order> Order(World world, CPos cell, int2 worldPixel, MouseInput mi)
+		{
+			if ((mi.Button == MouseButton.Left && mi.Event == MouseInputEvent.Down) || (mi.Button == MouseButton.Right && mi.Event == MouseInputEvent.Up))
+				return InnerOrder(world, cell, worldPixel, mi);
+
+			return Enumerable.Empty<Order>();
+		}
+
+		public abstract void Tick(World world);
```

Comment:
Can we make these `protected`, and then explicitly implement `IOrderGenerator.Tick` etc to call them?
This would go well with our long term goal of hiding internal implementation details via explicit interface implementations.

Reply:
Not sure if i understand this right something like this?
```suggestion
	public abstract class OrderGenerator : IOrderGenerator
	{
	        ...
		protected abstract void Tick(World world);
		protected abstract IEnumerable<IRenderable> Render(WorldRenderer wr, World world);
		protected abstract IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr, World world);
		protected abstract string GetCursor(World world, CPos cell, int2 worldPixel, MouseInput mi);
                ...
	}
	
	public class SelectGenericPowerTarget : OrderGenerator, IOrderGenerator
	{
                 ...
		  protected override void IOrderGenerator.Tick(World world)
		  {

		  }
                  ...
	}
```

Which is not possible

Reply:
`protected override void Tick` on `SelectGenericPowerTarget` etc, and then `void ITick.Tick(Actor self) { Tick(self); }` on `OrderGenerator`.